internal/pkg/build/build.go

```diff
@@ -292,10 +308,13 @@ func runBuildEngine(b *types.Bundle) error {
 	return starterCmd.Run()
 }
 
-func getcp(def types.Definition, libraryURL, authToken string) (ConveyorPacker, error) {
+func getcp(def types.Definition, libraryURL, authToken string, unauthenticatedBuild, localVerifyBuild bool) (ConveyorPacker, error) {
```

Comment:
I'm going to guess `ConveyorPacker` is an interface.

The interface should be extended so that it has two new methods (say `AllowUnauthenticated()` and something else for which I don't have a good suggestion). The Conveyors (or Packers) that don't support unauthenticated operations should return an error and the ones that do should default to authenticated and go back to unauthenticated when this function is called.

So the code would call this "factory" function `getcp` to obtain a ConveyorPacker and then the client code should call `AllowUnauthenticated()` as appropriate.